analyzer/src/main/kotlin/curation/Sw360PackageCurationProvider.kt

```diff
@@ -0,0 +1,133 @@
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.curation
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.eclipse.sw360.antenna.http.HttpClientFactoryImpl
+import org.eclipse.sw360.antenna.http.config.HttpClientConfig
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360Connection
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360ConnectionFactory
+import org.eclipse.sw360.antenna.sw360.client.config.SW360ClientConfig
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.attachments.SW360AttachmentType
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.releases.SW360ClearingState
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.releases.SW360Release
+
+import org.ossreviewtoolkit.analyzer.PackageCurationProvider
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.HashAlgorithm
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.PackageCuration
+import org.ossreviewtoolkit.model.PackageCurationData
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.config.Sw360StorageConfiguration
+import org.ossreviewtoolkit.model.jsonMapper
+
+/**
+ * A [PackageCurationProvider] for curated package meta-data from the configured SW360 instance using the REST API.
+ */
+class Sw360PackageCurationProvider(sw360Configuration: Sw360StorageConfiguration?) : PackageCurationProvider {
+
+    init {
+        requireNotNull(sw360Configuration) {
+            "SW360 is not configured. To use curations from SW360, configure it in the ORT configuration file."
+        }
+    }
+
+    private val sw360Connection = createSw360Connection(
+        sw360Configuration!!,
+        jsonMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    )
+
+    override fun getCurationsFor(pkgId: Identifier): List<PackageCuration> {
+        val name = listOfNotNull(pkgId.namespace, pkgId.name).joinToString("/")
+        val sw360ReleaseClient = sw360Connection.releaseAdapter
+
+        return sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkgId.version)
+            .flatMap { sw360ReleaseClient.enrichSparseRelease(it) }
+            .filter { it.sw360ClearingState == SW360ClearingState.APPROVED }
+            .map { sw360Release ->
+                listOf(
+                    PackageCuration(
+                        id = pkgId,
+                        data = PackageCurationData(
+                            declaredLicenses = sw360Release
+                                .embedded
+                                ?.licenses
+                                ?.filterNotNull()
+                                ?.map { license -> license.shortName }
+                                ?.toSortedSet() ?: sortedSetOf(),
```

Comment:
How about

    declaredLicenses = sw360Release.embedded?.licenses
        ?.mapNotNullTo(sortedSetOf()) { it?.shortName },